src/main/java/org/jabref/gui/preferences/externalfiletypes/ExternalFileTypesTabViewModel.java

```diff
@@ -54,25 +59,59 @@ public void resetToDefaults() {
         fileTypes.sort(Comparator.comparing(ExternalFileTypeItemViewModel::getName));
     }
 
-    public void addNewType() {
+    public boolean addNewType() {
         ExternalFileTypeItemViewModel item = new ExternalFileTypeItemViewModel();
-        fileTypes.add(item);
         showEditDialog(item, Localization.lang("Add new file type"));
+
+        if (!isValidExternalFileType(item)) {
+            return false;
+        }
+
+        fileTypes.add(item);
+        return true;
     }
 
     public ObservableList<ExternalFileTypeItemViewModel> getFileTypes() {
         return fileTypes;
     }
 
-    private void showEditDialog(ExternalFileTypeItemViewModel item, String dialogTitle) {
+    protected void showEditDialog(ExternalFileTypeItemViewModel item, String dialogTitle) {
         dialogService.showCustomDialogAndWait(new EditExternalFileTypeEntryDialog(item, dialogTitle));
     }
 
-    public void edit(ExternalFileTypeItemViewModel type) {
-        showEditDialog(type, Localization.lang("Edit file type"));
+    public boolean edit(ExternalFileTypeItemViewModel type) {
+        ExternalFileTypeItemViewModel typeToModify = new ExternalFileTypeItemViewModel(type.toExternalFileType());
+        showEditDialog(typeToModify, Localization.lang("Edit file type"));
+
+        if (!isValidExternalFileType(typeToModify)) {
+            return false;
+        }
+
+        fileTypes.remove(type);
+        fileTypes.add(typeToModify);
+        return true;
     }
 
     public void remove(ExternalFileTypeItemViewModel type) {
         fileTypes.remove(type);
     }
+
+    public boolean isValidExternalFileType(ExternalFileTypeItemViewModel item) {
+        // Check that there are no empty strings.
+        if (item.getName().isEmpty() || item.extensionProperty().get().isEmpty() || item.mimetypeProperty().get().isEmpty()) {
+            LOGGER.info("One of the fields is empty or invalid!");
+            return false;
+        }
+
+        // check extension need to be unique in the list
+        String newExt = item.extensionProperty().get();
+        for (ExternalFileTypeItemViewModel fileTypeItem : fileTypes) {
+            if (newExt.equalsIgnoreCase(fileTypeItem.extensionProperty().get())) {
+                LOGGER.info("File Extension exists already!");
```

Comment:
Do we really need this info to the end user? Isn't it just `debug`?

(And no exclamation mark)